contracts/LoanCore.sol

```diff
@@ -368,12 +407,20 @@ contract LoanCore is
         if (protocolFee > 0) _feesWithdrawable[address(this)] += protocolFee;
         if (affiliateFee > 0) _feesWithdrawable[affiliate] += affiliateFee;
 
-        // Delete the receipt
-        delete noteReceipts[loanId];
-
-        // Burn the note
+        // Get owner of the LenderNote
         address lender = lenderNote.ownerOf(loanId);
-        lenderNote.burn(loanId);
+
+        // if the loan has been completely repaid and no more repayments are expected
+        if (loans[loanId].balance == 0) {
```

Comment:
Seems more intuitive to read `LoanState` here? Or if we want to use `balance == 0` as truth, can we remove the `LoanState` enum altogether and save some gas?

Reply:
No I think checking LoanState is the best indicator here. Made this change